### PR TITLE
chore(docs): add guide for abstract types

### PR DESCRIPTION
**Pull request by jasonkuhrt**
TODO 

- [x] show how to pick a strategy in the config

### CODE REVIEW

File: docs/content/014-guides/021-abstract-types.md [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    }
+    ... on Song {
+      album
+      url
+    }
+  }
+}
+```
+
+In order for this builtin polymorphic system to work, your GrpahQL API must ultimately annotate the outgoing data with a property to discriminate results by on the client side. The spec states that this discriminant property is `__typename`.
```
Comment (Weakky) [inline docs/content/014-guides/021-abstract-types.md:78]:
```suggestion
In order for this builtin polymorphic system to work, your GraphQL API must ultimately annotate the outgoing data with a property to discriminate results by on the client side. The spec states that this discriminant property is `__typename`.
```

**Review by Weakky — APPROVED** [review]
great woooooork :100

File: docs/content/014-guides/021-abstract-types.md [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+   - The model type includes `__typename` property whose type is a string literal matching the GraphQL object name (case sensitive).
+   - The fields where the union type is used include `__typename` in the returned model data
+
+#### Multiple Strategies at Runtime
+
+It is possible to have all three strategies enabled and thus use at once. In that case the following runtime precedence rules apply. Using a strategy here means the implementations of others of lower priority are discarded.
+
+1. `resolveType`
+2. `__typename`
+3. `isTypeOf`
```
Comment (Weakky) [inline docs/content/014-guides/021-abstract-types.md:330]: There no such precedence in the current implementation as we've now moved to the `features` switch. If you enabled all of them, they'll just all error at the same time until you satisfy the requirement of one of the enabled strategy

Comment (jasonkuhrt) [reply docs/content/014-guides/021-abstract-types.md:330]: What about a case where multiple strategies are implemented though? Its useful to know what will happen at runtime no?

File: docs/content/014-guides/021-abstract-types.md [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+
+| Questions                                                                             | If affirmative, then maybe                                                                        |
+| ------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------- |
+| Is my schema large? Do I have many collaborators?                                     | Modular (`__typename`, `isTypeOf`)                                                                |
+| Is my schema simple? Do I develop alone?                                              | Centralized (`resolveType`)                                                                       |
+| Do I keep a discriminant property in the databse already?                             | `__typename` in your model layer if you have one                                                  |
+| Do I have objects that are part of multiple unions types and/or implement interfaces? | `__typename` or `isTypeOf` to avoid repeated logic across multiple `resolveType` implementations) |
```
Comment (Weakky) [inline docs/content/014-guides/021-abstract-types.md:347]: While you explain the different strategy very well, unless I'm mistaking, you do not explain how to pick one strategy using the global config: `features: { abstractTypes: { isTypeOf|resolveType|backingType?: boolean } }`

File: docs/content/014-guides/021-abstract-types.md [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+type Song implements Media {
+  album: String
+}
+
+type Query {
+  search(pattern: String): Media
+}
+```
+
+When a client sends a `serarch` query they will be able to select common fields as specified by the interface. Note this is unlike unions in which GraphQL assumes zero overlap between members. In contrast in GraphQL interfaces signify an intersection of fields between implementors and thus can be selected unqualified.
```
Comment (Weakky) [inline docs/content/014-guides/021-abstract-types.md:385]:
```suggestion
When a client sends a `search` query they will be able to select common fields as specified by the interface. Note this is unlike unions in which GraphQL assumes zero overlap between members. In contrast in GraphQL interfaces signify an intersection of fields between implementors and thus can be selected unqualified.
```

File: docs/content/014-guides/021-abstract-types.md (hunk 2 above)
Comment (Weakky) [reply docs/content/014-guides/021-abstract-types.md:330]: Oh, I see, this is how GraphQL works at runtime, nvm! 👍

**Pull request merged**